src/main/scala/org/constellation/snapshotstreaming/App.scala

```diff
@@ -1,328 +1,45 @@
 
 object App extends IOApp {
-  private val logger = Slf4jLogger.getLogger[IO]
+  def mkLogger = Slf4jLogger.create[IO]
   private val configuration = new Configuration
-  private val storedSnapshotMapper = new StoredSnapshotMapper
-  private val snapshotInfoMapper = new SnapshotInfoMapper
-  private val balanceValidator = new BalanceValidator[IO](Map.empty)
 
   def run(args: List[String]): IO[ExitCode] =
-    main(args).compile.drain
-      .flatTap(_ => logger.debug("Done!"))
-      .map(_ => ExitCode.Success)
-      .handleErrorWith(e => logger.error(e)(e.getMessage).map(_ => ExitCode.Error))
-
-  def main(args: List[String]): Stream[IO, Unit] =
-    for {
-      _ <- Stream.eval(KryoSerializer.init[IO])
-      s3Client <- Stream.bracket(
-        IO(
-          AmazonS3ClientBuilder
-            .standard()
-            .withRegion(configuration.bucketRegion)
-            .build()
-        )
-      )(c => IO(c.shutdown()))
-
-      elasticSearchDAO = ElasticSearchDAO[IO](
-        storedSnapshotMapper,
-        snapshotInfoMapper,
-        configuration
-      )
-
-      s3DAOs = configuration.bucketNames.map(S3DAO[IO](s3Client)(_))
-
-      _ <- if (configuration.getGenesis)
-        getAndSendGenesisToES(s3DAOs, elasticSearchDAO)
-      else
-        Stream.eval(LiftIO[IO].liftIO(().pure[IO]))
-
-      _ <- if (configuration.getSnapshots)
-        getAndSendSnapshotsToES(s3DAOs, elasticSearchDAO)
-      else
-        Stream.eval(LiftIO[IO].liftIO(().pure[IO]))
-    } yield ()
-
-  private def getAndSendGenesisToES[F[_]: Timer: RaiseThrowable: ConcurrentEffect: Parallel](
-    s3DAOs: List[S3DAO[F]],
-    elasticSearchDAO: ElasticSearchDAO[F]
-  ): Stream[F, Unit] =
-    for {
-      genesis <- getGenesisFromS3(s3DAOs)
-      _ <- putGenesisToES(elasticSearchDAO)(genesis)
-    } yield ()
-
-  private def getAndSendSnapshotsToES[F[_]: Timer: RaiseThrowable: ConcurrentEffect: Parallel: ContextShift](
-    s3DAOs: List[S3DAO[F]],
-    elasticSearchDAO: ElasticSearchDAO[F]
-  ): Stream[F, Unit] =
-    for {
-      s3Object <- getFromS3(s3DAOs)
-
-      _ <- s3Object
-        .map(putToES(elasticSearchDAO))
-        .getOrElse(Stream.emit(()))
-
-      _ <- s3Object
-        .map(r => saveLastStreamedHeightToFile(r.height))
-        .getOrElse(Stream.emit(()))
-
-      _ <- s3Object
-        .map(r => Stream.eval(LiftIO[F].liftIO(balanceValidator.validate(r))))
-        .getOrElse(Stream.emit(()))
-    } yield ()
-
-  private def getGenesisFromS3[F[_]: Concurrent: Timer: RaiseThrowable](
-    s3DAOs: List[S3DAO[F]]
-  ): Stream[F, S3GenesisDeserializedResult] = {
-    s3DAOs match {
-      case List(bucket) =>
-        bucket.getGenesis().handleErrorWith { e =>
-          Stream.eval(
-            LiftIO[F].liftIO(
-              logger.error(e)(s"[S3-Genesis ->] ERROR. While getting Genesis from bucket=${bucket.getBucketName}.")
-            )
-          ) >> Stream.raiseError(e)
+    Slf4jLogger.create[IO].flatMap { implicit logger =>
```

Comment:
I left a comment here previously but looks like it has been ignored. Slf4jLogger creation part is duplicated (see: def mkLogger)